bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/inbox/InboxFilterCriteria.java

```diff
@@ -26,61 +27,137 @@
     private final String bindingId;
     private final ThingTypeUID thingTypeUID;
     private final ThingUID thingUID;
+    private final String representationValue;
```

Comment:
I find this a bit limiting. What if a consumer wants to filter for some other property (e.g. MANUFACTURER) or so?

@ViToni recently introduced a nice new stream based filtering for registries, which has been adopted in the RuleRegistry by defining [RulePredicates](https://github.com/eclipse/smarthome/blob/master/bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RulePredicates.java). This mechanism was meant to be used from now on throughout ESH, so imho it would make a lot of sense to go for it here as well, instead of introducing all those `xxxFilter()` methods, wdyt?
This would allow us to simply have a "property predicate", where the consumer can choose whether he wants to use the representation property or any other. There would thus be no need for a dedicated field for just this purpose here.